### ISSUE TITLE
Batch normalization 

### ISSUE THREAD

**Issue by RadjaHachilif** [intel/mkl-dnn#349]
Hello,
i need to use batch normalization primitive without calculating mean and variance 
i found this syntax but im not sure how it operates on a 4D input ( like does it calculate the mean of all values  or using values within a hyperplane )
batch_normalization_forward (const primitive_desc &aprimitive_desc, const primitive::at &src, const memory &dst)

can you please explain for me how this primitive works ?
Thank you
-----------------------------------------------------------------------------

### Environment
Intel MKL-DNN includes hardware-specific optimizations and may behave
differently on depending on the compiler and build environment. Include
the following information to help reproduce the issue:
* CPU make and model (try `lscpu`; if your `lscpu` does not list CPU flags,
  try running `cat /proc/cpuinfo | grep flags | sort -u`)
* OS version (`uname -a`)
* Compiler version (`gcc --version`)
* MKLROOT value (`echo MKLROOT=$MKLROOT`)
* CMake version (`cmake --version`)
* CMake output log
* git hash (`git log -1 --format=%H`)

### Steps to reproduce
Please check that the issue is reproducible with the latest revision on
master. Include all the steps to reproduce the issue. A short C/C++ program
or modified unit tests demonstrating the issue will greatly help
with the investigation.

### Actual behavior
Describe the behavior you see.

### Expected behavior
Describe the behavior you expect.

**Comment by emfomenk** [comment]
Hi Radja,

Yeah, our C++ documentation is somewhat incomplete... The behavior that you want to get is controlled at batch normalization descriptor initialization using the following [C API](https://intel.github.io/mkl-dnn/group__c__api__batch__normalization.html#ga3d3498ea608a63f6ed09a825414d02af) / [C++ API](https://intel.github.io/mkl-dnn/structmkldnn_1_1batch__normalization__forward_1_1desc.html#aac8dc676e445e93676dffac6ef4390a8). There you pass `flags` that might take one or several bit-masks described here: [C API](https://intel.github.io/mkl-dnn/group__c__api__types__generic.html#ga5b1f166df8eea3a306f9226baf6ca3a3) / [C++](https://intel.github.io/mkl-dnn/group__cpp__api__enums.html#ga344b5c240a7392585aba615ff4d0d7f1).

So if you don't want Intel MKL-DNN to compute the mean and variance (but rather provide them by yourself) you need to set flags to `mkldnn_use_global_stats`.

A simple example can be found in our tests, e.g. see [here](https://github.com/intel/mkl-dnn/blob/v0.17-rc/tests/gtests/test_batch_normalization.cpp#L312).

**Comment by RadjaHachilif** [comment]
Thank you, this really helped.
